File: typescript-connector/src/models/ProductOption.ts
// Class from DFC Business Ontology: #ProductOption
import { SemanticObject } from "../core/SemanticObject.js";
import { WhatSubject, type WhatSubjectParams } from "./WhatSubject.js";
import type { ProductOptionValue } from "./ProductOptionValue.js";

export interface ProductOptionParams extends WhatSubjectParams {
  hasReferenceProductOptionValue?: ProductOptionValue;
}
export class ProductOption extends WhatSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:ProductOption";
  }

  hasReferenceProductOptionValue?: ProductOptionValue;

  constructor(
    semanticId: string,
    params?: ProductOptionParams,
  ) {
    super(semanticId, params);
    this.hasReferenceProductOptionValue = params?.hasReferenceProductOptionValue;
    this.semanticType = ProductOption.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:hasReferenceProductOptionValue", () => this.hasReferenceProductOptionValue);
  }
  static {
    SemanticObject.typeRegistry.set(ProductOption.SEMANTIC_TYPE, ProductOption);
  }
}

File: ruby-gem/lib/models/product_option.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #ProductOption
require_relative '../core/semantic_object'
require_relative 'what_subject'


module DfcLinkmlConnector
  module Models
    class ProductOption < WhatSubject
      SEMANTIC_TYPE = "dfc-b:ProductOption".freeze

      # @return [ProductOptionValue]
      attr_accessor :reference_product_option_value

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, referenceProductOptionValue: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, referenceProductOptionValue: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @reference_product_option_value = referenceProductOptionValue
        self.semanticType = "dfc-b:ProductOption"
        registerSemanticProperty("dfc-b:hasReferenceProductOptionValue", &method("reference_product_option_value")).valueSetter = method("reference_product_option_value=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
